tests/test_communication_manager.py

```diff
@@ -72,7 +96,33 @@ def test_communication_manager_can_initialize_and_train_model():
         scheduler.runners_run_next_jobs()
         time.sleep(0.1)
     assert len(scheduler.processed) == 2
-
+    while len(scheduler.processed) == 2:
```

Comment:
This should timeout after a short amount of time, so that tests don't hang in case stuff breaks.